### PR TITLE
Fix issues found with phpstan level 0

### PR DESCRIPTION
**Pull request by vladyslavstartsev**
see https://github.com/sonata-project/SonataAdminBundle/pull/6154#issuecomment-646969844 for details
Removed classes that just can't be used.

I am targeting this branch, because this will not peoples code.

## Changelog

```markdown
### Changed
- `ModelsToArrayTransformer` to not refer to `ModelChoiceList`
### Removed
- class `ModelChoiceList` (file src/Form/ChoiceList/ModelChoiceList.php)
- class `LegacyModelsToArrayTransformer` (file src/Form/DataTransformer/LegacyModelsToArrayTransformer.php)
- remove all references to `jms/di-extra-bundle` and it's usages in Code
```

### CODE REVIEW

File: phpstan.neon.dist [hunk 1]
```diff
@@ -7,7 +7,6 @@ parameters:
         - src
     excludes_analyse:
         # Class Symfony\Component\Form\Extension\Core\ChoiceList\SimpleChoiceList not found. I can't even find this class too
-        - src/Form/ChoiceList/ModelChoiceList.php
         - src/Form/DataTransformer/LegacyModelsToArrayTransformer.php
```
Comment (VincentLanglet) [inline phpstan.neon.dist:10]: You can remove this line too

Comment (vladyslavstartsev) [reply phpstan.neon.dist:10]: yeah, forgot it. Thanks!

**Comment by VincentLanglet** [comment]
> - remove referrence to `LegacyModelsToArrayTransformer` in `phpstan` config (since it's not used anymore

I don't think this is needed in the Changelog, it's pedantic.

> ### Fixed
> - remove dead code

You already listed the removed class, so I would remove this line.

File: phpstan.neon.dist [hunk 2]
```diff
@@ -7,10 +7,6 @@ parameters:
         - src
     excludes_analyse:
         # Class Symfony\Component\Form\Extension\Core\ChoiceList\SimpleChoiceList not found. I can't even find this class too
```
Comment (VincentLanglet) [inline phpstan.neon.dist:9]: You forgot the comment ^^'

**Comment by vladyslavstartsev** [comment]
by the way, the ignorance of `src/Annotation/Admin.php` is due to `jms/di-extra-bundle`

But when I try to install it, I get error that looks something like this 
```text
jms/di-extra-bundle 1.9.2 requires symfony/framework-bundle ~2.3|~3.0 -> satisfiable by symfony/framework-bundle
```
due to `composer.json` having requirements `"symfony/framework-bundle": "^4.3",`
so this class, also, will never be executed

Am I right? Or missing something?

**Comment by vladyslavstartsev** [comment]
@VincentLanglet forgot to mention

**Comment by VincentLanglet** [comment]
I agree with you @vladyslavstartsev 
I don't see how someone could have both sonata and jms/di-extra-bundle on the same project.
You can remove it too.

File: src/DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -159,10 +159,6 @@ public function getConfigTreeBuilder()
                             ->defaultFalse()
                             ->info('Enable locking when editing an object, if the corresponding object manager supports it.')
                         ->end()
-                        ->booleanNode('enable_jms_di_extra_autoregistration') // NEXT_MAJOR: remove this option
```
Comment (VincentLanglet) [inline src/DependencyInjection/Configuration.php:162]: What if someone has this option in his config ? 

The class couldn’t be used but theorically someone could have this in his config and the update will throw an error. Isn’t it ?

Comment (vladyslavstartsev) [reply src/DependencyInjection/Configuration.php:162]: I have checked, no it will not thow error. But I may be wrong. 
"Works on my machine" ©

**Comment by greg0ire** [comment]
Thanks @vladyslavstartsev !

**Pull request merged**